tests/std/tests/GH_003883_replace_asan_failure/test.cpp

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <string>
+
+int main() {
+    std::string t = "0123456789ABCDEF"; // large string
+    t.replace(0, 30, 7, 'A');

```

Comment:
In general, after calling a modifying function, we should verify the final state. This helps catch all sorts of bugs, including those we might not have expected.